src/hooks/wallet/MetaMaskWallet.ts

```diff
@@ -180,6 +204,75 @@ export class MetaMaskWallet implements WalletInterface {
     return tokenContract
   }
 
+  public async transferNativeTokens(
+    receiver: string,
+    amount: string | BigNumber,
+    asset: AssetInfo,
+    sourceChainName: string
+  ): Promise<MetamaskTransferEvent> {
+    const response: MetamaskTransferEvent = {
+      txHash: "",
+      tokenContractAddress: "",
+      receiver,
+      amount: "",
+      error: "",
+      blockNumber: "",
+    }
+
+    let userAddress = await this.getAddress()
+
+    const env: string =
+      process.env.REACT_APP_STAGE === "mainnet" ? "mainnet" : "testnet"
+    const contractAddress: string =
+      convertAndDepositContractAddress[env][
+        sourceChainName?.toLowerCase() || ""
+      ]
+    const ethersContract = new ethers.Contract(
+      contractAddress,
+      convertAndDepositAbi,
+      this.signer
+    )
+
+    response.tokenContractAddress = ""
+    try {
+      receiver = ethers.utils.getAddress(receiver)
+    } catch {
+      response.error += `, Invalid address: ${receiver}, `
+    }
+
+    try {
+      amount = ethers.utils.parseUnits(amount as string, asset.decimals)
+      if (amount.isNegative()) {
+        throw new Error()
+      }
+    } catch (e) {
+      console.error(`Invalid amount: ${amount}` + e)
+      response.error += `, Invalid amount: ${amount}`
+    }
+
+    const balance = await this.provider.getBalance(userAddress)
+    console.log("balance and amount", balance, amount)
+
+    if (balance.lt(amount)) {
+      let amountFormatted = ethers.utils.formatUnits(amount, asset.decimals)
+      let balanceFormatted = ethers.utils.formatUnits(balance, asset.decimals)
+      console.error(
+        `Insufficient balance receiver send ${amountFormatted} (You have ${balanceFormatted})`
+      )
+      response.error += `, Insufficient balance receiver send ${amountFormatted} (You have ${balanceFormatted})`
+    }
```

Comment:
I think it's cleaner to validate everything at the very beginning of the function. 🙂